packages/cerebral/src/Controller.js

```diff
@@ -183,21 +183,17 @@ class Controller extends FunctionTree {
           throw error
         }
 
-        if (typeof signalCatch === 'function') {
-          const signalChain = signalCatch(error)
+        if (signalCatch instanceof Map) {
+          for (let [errorType, signalChain] of signalCatch) {
+            if (error instanceof errorType) {
+              this.runSignal('catch', signalChain, error.payload)
 
-          if (signalChain) {
-            this.runSignal('catch', signalChain, error.payload)
-          } else {
-            throw error
+              return
```

Comment:
@christianalfoni Yeah sorry `break` would only break out of the for...of loop, not the whole function.